composables/useCollectionAttributes.ts

```diff
@@ -1,19 +1,25 @@
 import { useQuery } from '@tanstack/vue-query'
-import type { NFT, Attribute } from '@/types'
+import type { Attribute } from '@/types'
+import type { NFT } from '@/components/massmint/types'
 
-export const useCollectionAttributes = (collectionId: ComputedRef<string | undefined>) => {
-  const { data: nftsListData } = useQuery<NFT[]>({
+type NftWithAttributes = { meta?: { attributes?: Attribute[] } }
+
+export const useCollectionAttributes = ({ collectionId, extraNfts }: { collectionId: ComputedRef<string | undefined>, extraNfts?: ComputedRef<NFT[]> }) => {
+  const { data: nftsListData } = useQuery<NftWithAttributes[]>({
     queryKey: ['nft-attributes-by-collection', collectionId],
     queryFn: async () =>
       (collectionId.value
-        ? (await useAsyncGraphql<{ nfts: NFT[] } >({
+        ? (await useAsyncGraphql<{ nfts: NftWithAttributes[] } >({
             query: 'nftAttributesListByCollection',
             variables: { id: collectionId.value },
           })).data.value.nfts
         : []),
   })
 
-  const nftsList = computed(() => nftsListData.value || [])
+  const nftsList = computed(() => (nftsListData.value || []).concat(
+    extraNfts?.value
+      .map(nft => ({ meta: {
+        attributes: nft.attributes?.filter(attr => String(attr.trait_type) && String(attr.value)).map(attr => ({ trait: attr.trait_type, value: String(attr.value) })) } })) || []))
```

Comment:
Nit: a little hard to read if everything is one line , id break line after .map .filter and format accordingly so that is easy on the eye